src/graphics/program-library.js

```diff
@@ -39,7 +43,7 @@ Object.assign(pc, function () {
         var shader = this._cache[key];
         if (!shader) {
             var shaderDefinition = generator.createShaderDefinition(gd, options);
-            shader = this._cache[key] = new pc.Shader(gd, shaderDefinition);
+            shader = this._cache[key] = new pc.Shader(gd, shaderDefinition, true);

```

Comment:
If it's better to do things with 'true', we should probably just make the the default behavior instead of changing the API and exposing the option?

Reply:
Intention was to minimise amount of affected components.

Reply:
If making 'true' (linking immediately) the default behavior yields superior results, I think it's preferable to update multiple internal components instead of modifying public API.